subsys/bluetooth/mesh/brg_cfg.c

```diff
@@ -0,0 +1,317 @@
+		brg_enabled = 0;
+		LOG_DBG("Cleared bridge enable state");
+		return 0;
+	}
+
+	err = bt_mesh_settings_set(read_cb, cb_arg, &brg_enabled, sizeof(brg_enabled));
+	if (err) {
+		LOG_ERR("Failed to set bridge enable state");
+		return err;
+	}
+
+	LOG_DBG("Restored sbren");
+
+	return 0;
+}
+
+/* Define a setting for storing enable state */
+BT_MESH_SETTINGS_DEFINE(brgen, "brgen", brgen_set);
+
+/* Set function for initializing bridging table rows from brg_tbl slist */
+static int brg_tbl_set(const char *name, size_t len_rd, settings_read_cb read_cb, void *cb_arg)
+{
+	if (len_rd == 0) {
+		memset(brg_tbl, 0, sizeof(brg_tbl));
+		brg_tbl_row_cnt = 0;
+		LOG_DBG("Cleared bridging table entries");
+		return 0;
+	}
+
+	int err = bt_mesh_settings_set(read_cb, cb_arg, &brg_tbl, sizeof(brg_tbl));
+
+	if (err) {
+		LOG_ERR("Failed to set bridging table entries");
+		return err;
+	}
+
+	LOG_DBG("Restored brg_tbl");
+
+	return 0;
+}
+
+/* Define a setting for storing briging table rows */
+BT_MESH_SETTINGS_DEFINE(brgbt, "brgbt", brg_tbl_set);
+#endif
+
+bool brg_cfg_enable_get(void)
+{
+	return brg_enabled;
+}
+
+int brg_cfg_enable_set(bool enable)
+{
+	if (brg_enabled == enable) {
+		return 0;
+	}
+
+	brg_enabled = enable;
+#if IS_ENABLED(CONFIG_BT_SETTINGS)
+	bt_mesh_settings_store_schedule(BT_MESH_SETTINGS_BRG_PENDING);
+#endif
+	return 0;
+}
+
+void bt_mesh_brg_en_pending_store(void)
+{
+#if CONFIG_BT_SETTINGS
+	char *path = "bt/mesh/brgen";
+	int err;
+
+	if (brg_enabled) {
+		err = settings_save_one(path, &brg_enabled, sizeof(brg_enabled));
+	} else {
+		err = settings_delete(path);
+	}
+
+	if (err) {
+		LOG_ERR("Failed to store %s value", path);
+	}
+#endif
+}
+
+void bt_mesh_brg_tbl_pending_store(void)
+{
+#if CONFIG_BT_SETTINGS
+	char *path = "bt/mesh/brgbt";
+	char path_bt[20];
+	int err;
+
+	if (!brg_tbl_row_cnt) {
+		err = settings_save_one(path, &brg_tbl, brg_tbl_row_cnt * sizeof(brg_tbl[0]));
+	} else {
+		err = settings_delete(path);
+	}
+
+	if (err) {
+		LOG_ERR("Failed to store %s value", path_bt);
+	}
+#endif
+}
+
+/* Remove the entry from the bridging table that corresponds with the netkey index
+ * of the removed subnet.
+ */
+static void brg_tbl_netkey_removed_evt(struct bt_mesh_subnet *sub, enum bt_mesh_key_evt evt)
+{
+	for (int i = 0; i < CONFIG_BT_MESH_BRG_TABLE_ITEMS_MAX; i++) {
+		if (brg_tbl[i].direction && (
+		    brg_tbl[i].net_idx1 == sub->net_idx ||
+		    brg_tbl[i].net_idx2 == sub->net_idx)) {
+			memset(&brg_tbl[i], 0, sizeof(brg_tbl[i]));
+			brg_tbl_compact();
+		}
+	}
+
+#if IS_ENABLED(CONFIG_BT_SETTINGS)
+	bt_mesh_settings_store_schedule(BT_MESH_SETTINGS_BRG_PENDING);
+#endif
+}
+
+/* Add event hook for key deletion event */
+BT_MESH_SUBNET_CB_DEFINE(sbr) = {
+	.evt_handler = brg_tbl_netkey_removed_evt,
+};
+
+int brg_cfg_init(void)
+{
+	memset(brg_tbl, 0, sizeof(brg_tbl));
+	brg_tbl_row_cnt = 0;
+	return 0;
+}
+
+int brg_cfg_tbl_reset(void)
+{
+	brg_cfg_init();
+
+	int err = settings_delete("bt/mesh/brgen");
```

Comment:
```suggestion
	int err = settings_delete("bt/mesh/brg_en");
```

Reply:
We don't want to keep increasing footprint of settings file name. This should be as small as possible.

Reply:
3 bytes increasing :) for better naming compatibility within the implementation. I'm pretty sure @HaavardRei's suggestion is not so bad to reject it.

Reply:
This is superficial ask. This goes onto the flash and you want to push additional stuff on flash just for readability of flash?  Will update it in next round.